defuse/src/accounts/mod.rs

```diff
@@ -0,0 +1,109 @@
+mod account;
+mod nonces;
+mod state;
+
+use std::collections::HashSet;
+
+pub use self::{account::*, nonces::*, state::*};
+
+use defuse_contracts::{
+    crypto::PublicKey,
+    defuse::accounts::AccountManager,
+    nep413::Nonce,
+    utils::{cache::PREDECESSOR_ACCOUNT_ID, prefix::NestPrefix, serde::wrappers::DisplayFromStr},
+};
+use near_sdk::{
+    borsh::BorshSerialize, near, store::IterableMap, AccountId, BorshStorageKey, IntoStorageKey,
+};
+
+use crate::{DefuseImpl, DefuseImplExt};
+
+#[near]
+impl AccountManager for DefuseImpl {
+    fn public_keys_of(&self, account_id: &AccountId) -> HashSet<PublicKey> {
+        self.accounts
+            .get(account_id)
+            .into_iter()
+            .flat_map(Account::iter_public_keys)
+            .copied()
+            .collect()
+    }
+
+    fn add_public_key(&mut self, public_key: PublicKey) {
+        self.accounts
+            .get_or_create(PREDECESSOR_ACCOUNT_ID.clone())
+            .add_public_key(&PREDECESSOR_ACCOUNT_ID, public_key);
```

Comment:
The method returns a value you do not use here. It's a point for simplifying the `add_public_key` method.